composer/models/bert/bert_hparams.py

```diff
@@ -5,130 +5,113 @@
 :class:`.BERTModel`."""
 
 from dataclasses import dataclass
-from typing import TYPE_CHECKING
+from typing import Dict, Optional
 
 import yahp as hp
 
-from composer.models.transformer_hparams import TransformerHparams
-from composer.utils import MissingConditionalImportError
-
-if TYPE_CHECKING:
-    from composer.models.bert import BERTModel
+from composer.core.types import JSON
+from composer.models.model_hparams import ModelHparams
+from composer.utils.import_helpers import MissingConditionalImportError
 
 __all__ = ["BERTForClassificationHparams", "BERTHparams"]
 
 
 @dataclass
-class BERTForClassificationHparams(TransformerHparams):
-    """`YAHP <https://docs.mosaicml.com/projects/yahp/en/stable/README.html>`_ classification interface for
-    :class:`.BERTModel`.
+class BERTHparams(ModelHparams):
+    """`YAHP <https://docs.mosaicml.com/projects/yahp/en/stable/README.html>`_ interface for :class:`.BERTModel`.
 
     Args:
-        pretrained_model_name (str): Pretrained model name to pull from Hugging Face Model Hub.
```

Comment:
Wait, one quick question: why did we get rid of this? @ishanashastri @A-Jacobson 

How would we expect the user to change the underlying BERT model on the HF hub without another `pretrained_model_name`?

Reply:
I removed these because I didn't see it being changed in any of the yaml configs and thought you were including it to enable having the same hparam class for multiple models. of course, I wasn't sure which is why you're tagged to review =). If changing the underlying bert model is something we want to enable I see two paths for that. either we make multiple factory functions `create_bert_base` `create_bert_large` or we just add the `pretrained_model_name` back in. no problem with either.

Reply:
yup, moving the `pretrained_model_name` arg back seems to make most sense without having to change the existing yamls